### PR TITLE
Added support for Ericsson MINI-LINK (MLTN) 66XX & 63XX

### PR DESCRIPTION
**Pull request by codekuu**
Added support for Ericsson MLTN 66XX and 63XX series
Device Types Added:

- ericsson_mltn63
- ericsson_mltn66

### CODE REVIEW

**Comment by codekuu** [comment]
@ktbyers hey! Do you have time to look at this?
(you told me to ping you)

**Comment by ktbyers** [comment]
@codekuu Something broke with respect to Netmiko and GitHub actions. I am looking into it.

**Comment by ktbyers** [comment]
Okay, CI-CD issues fixed. Now I can review/check this.

**Comment by ktbyers** [comment]
@codekuu Did you run the Netmiko config and show tests against your code?

If so, can you post the output of your testing into the PR.

https://github.com/ktbyers/netmiko/blob/develop/TESTING.md

Thanks.

File: netmiko/ericsson/ericsson_mltn.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+    def exit_config_mode(self, exit_config: str = "exit", pattern: str = "#") -> str:
+        return super().exit_config_mode(exit_config=exit_config, pattern=pattern)
+
+    def save_config(
+        self, cmd: str = "write", confirm: bool = False, confirm_response: str = ""
+    ) -> str:
+        """Save Config for Ericsson Minilink"""
+        return super().save_config(
+            cmd=cmd, confirm=confirm, confirm_response=confirm_response
+        )
```
Comment (ktbyers) [inline netmiko/ericsson/ericsson_mltn.py:148]: See comments below about save_config and commit methods.

**Review by ktbyers — CHANGES_REQUESTED** [review]

**Comment by ktbyers** [comment]
Also is there any paging of output i.e. --more-- (and a way to disable this behavior)?

File: netmiko/ericsson/ericsson_mltn.py [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+
+class EricssonMinilinkBase(BaseConnection):
+    """Ericsson MiniLink Base class"""
+
+    def __init__(
+        self,
+        default_ssh_username: str = "cli",
+        login_timeout: int = 20,
+        login_sleep_time: float = 0.1,
+        login_sleep_multiplier: int = 5,
```
[the remark this answers is not in the document]
Comment (codekuu) [reply netmiko/ericsson/ericsson_mltn.py:19]: The problem with Ericsson MLTN is that the user you ssh with is not the same user that you sign in with.
So the ssh connection will be OK with cli@ but another user when the login prompt comes up.

**Comment by codekuu** [comment]
> Also is there any paging of output i.e. --more-- (and a way to disable this behavior)?
@ktbyers 

No there is nothing like that, and I am not able to get the testing suite running correctly.
Having issues when running the py.test...
```
/netmiko/tests$ py.test -v tests/test_netmiko_show.py --test_device ericsson_mltn66
ImportError while loading conftest '/netmiko/tests/conftest.py'.
conftest.py:8: in <module>
    from netmiko import ConnectHandler, FileTransfer, InLineTransfer, SSHDetect
E   ModuleNotFoundError: No module named 'netmiko'
```
I do not have netmiko installed when running the tests.

However, I have tested this module for both Ericsson MLTN 66XX and 63XX series and works perfectly, both with salt and without salt.

File: netmiko/ericsson/ericsson_mltn.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ktbyers) [reply netmiko/ericsson/ericsson_mltn.py]: Is the other user always `cli` or does that change?

**Comment by ktbyers** [comment]
@codekuu I will see if I can help with your testing issue, but might be tonight or tomorrow.

File: netmiko/ericsson/ericsson_mltn.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (codekuu) [reply netmiko/ericsson/ericsson_mltn.py]: As of my understanding it is default and not changeable so yes, it is always `cli`.
So I removed the `default_ssh_username` part from the code and hardcoded it.

**Comment by codekuu** [comment]
> @codekuu I will see if I can help with your testing issue, but might be tonight or tomorrow.

Awesome, thank you!

**Comment by ktbyers** [comment]
@codekuu Okay, here is the process for setting up the Netmiko environment:

https://github.com/ktbyers/netmiko/blob/develop/COMMON_ISSUES.md#installing-netmiko-develop-branch-with-poetry-support

You will need to change your branch to your `ericsson-mltn-support` branch, but the rest of the process should be essentially the same.

Let me know if you run into any issues.

**Comment by codekuu** [comment]
Weird that the mypy linting fails, works for me locally:
```
codekuu:/netmiko/netmiko/ericsson$ mypy --version
mypy 1.0.0 (compiled: yes)

codekuu:/netmiko/netmiko/ericsson$ mypy .
Success: no issues found in 3 source files
```

**Comment by codekuu** [comment]
Thanks, totally missed the mypy configuration. Everything is now in the green.

**Comment by ktbyers** [comment]
Superseded by:

https://github.com/ktbyers/netmiko/pull/3189

**Pull request closed without merging**